roles/validate_inventory/tasks/day2.yml

```diff
@@ -1,5 +1,7 @@
-- name: Check for day2_discovery_iso_name if required
+- name: Check for variables if required
   assert:
     that:
       - day2_discovery_iso_name is defined
+      - use_mce_for_day2 is defined
```

Comment:
Sorry when I looked earlier I didn't notice that the "when"s should have been different. I think it should be. 

```
- name: Check for variables if required
  assert:
    that:
      - day2_discovery_iso_name is defined
  when: (groups['day2_workers'] | default([])) | length > 0

- name: Check for variables if required
  assert:
    that:
      - day2_add_worker_namespace is defined
  when: 
    - (groups['day2_workers'] | default([])) | length > 0
    - use_mce_for_day2 is defined and (use_mce_for_day2 | bool)
```

Reply:
done